From 6158bebd37ba443c06aeb3ed9d6d44240592d418 Mon Sep 17 00:00:00 2001
From: Lorenzo Moneta <Lorenzo.Moneta@cern.ch>
Date: Tue, 1 Oct 2013 16:26:29 +0200
Subject: [PATCH] Fix for changes done with the vc merge:

 - Add a Scalar typedef definition in the TVectors  in order to use the ROOT::Math::VectorUtil functions
 - Fix a bug introduced in the Choleski decomposition for making it compiling with Vc. This fixes the testInversion test
---
 math/physics/inc/TLorentzVector.h      | 2 ++
 math/physics/inc/TVector2.h            | 2 ++
 math/physics/inc/TVector3.h            | 2 ++
 math/smatrix/inc/Math/CholeskyDecomp.h | 2 +-
 4 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/math/physics/inc/TLorentzVector.h b/math/physics/inc/TLorentzVector.h
index f5c58ba..d761021 100644
--- a/math/physics/inc/TLorentzVector.h
+++ b/math/physics/inc/TLorentzVector.h
@@ -44,6 +44,8 @@ private:
 
 public:
 
+   typedef Double_t Scalar;   // to be able to use it with the ROOT::Math::VectorUtil functions
+
    enum { kX=0, kY=1, kZ=2, kT=3, kNUM_COORDINATES=4, kSIZE=kNUM_COORDINATES };
    // Safe indexing of the coordinates when using with matrices, arrays, etc.
 
diff --git a/math/physics/inc/TVector2.h b/math/physics/inc/TVector2.h
index 20778eb..8c612bd 100644
--- a/math/physics/inc/TVector2.h
+++ b/math/physics/inc/TVector2.h
@@ -28,6 +28,8 @@ protected:
 //------------------------------------------------------------------------------
 public:
 
+   typedef Double_t Scalar;   // to be able to use it with the ROOT::Math::VectorUtil functions
+
    TVector2 ();
    TVector2 (Double_t *s);
    TVector2 (Double_t x0, Double_t y0);
diff --git a/math/physics/inc/TVector3.h b/math/physics/inc/TVector3.h
index a6df2e0..437f94c 100644
--- a/math/physics/inc/TVector3.h
+++ b/math/physics/inc/TVector3.h
@@ -28,6 +28,8 @@ class TVector3 : public TObject {
 
 public:
 
+   typedef Double_t Scalar;   // to be able to use it with the ROOT::Math::VectorUtil functions
+
    TVector3();
 
    TVector3(Double_t x, Double_t y, Double_t z);
diff --git a/math/smatrix/inc/Math/CholeskyDecomp.h b/math/smatrix/inc/Math/CholeskyDecomp.h
index ca38bf6..759a2db 100644
--- a/math/smatrix/inc/Math/CholeskyDecomp.h
+++ b/math/smatrix/inc/Math/CholeskyDecomp.h
@@ -527,7 +527,7 @@ namespace CholeskyDecompHelpers {
          dst[13] = (src(4,3) - dst[6] * dst[10] - dst[7] * dst[11] - dst[8] * dst[12]) * dst[9];
          dst[14] = src(4,4) - (dst[10]*dst[10]+dst[11]*dst[11]+dst[12]*dst[12]+dst[13]*dst[13]);
          if (dst[14] <= F(0.0)) return false;
-         else dst[14] = std::sqrt(F(.1) / dst[14]);
+         else dst[14] = std::sqrt(F(1.0) / dst[14]);
          return true;
       }
    };
-- 
1.8.4.2

